xchange-idex/src/main/java/org/knowm/xchange/idex/IdexExchange.java

```diff
@@ -0,0 +1,228 @@
+    allCurrenciesStatic.forEach(
+        (key, value) ->
+            linkedHashMap.put(
+                Currency.getInstance(key),
+                new IdexCurrencyMeta(
+                    0, ZERO, value.getAddress(), value.getName(), value.getDecimals())));
+    RateLimit[] publicRateLimits = {};
+    return new ExchangeMetaData(
+        currencyPairs, linkedHashMap, publicRateLimits, publicRateLimits, Boolean.FALSE);
+  }
+
+  private IdexAccountService idexAccountService;
+  private IdexTradeService idexTradeService;
+  private IdexMarketDataService idexMarketDataService;
+  private ReturnNextNonceApi nextNonceApi;
+
+  public IdexExchange() {}
+
+  public ReturnNextNonceApi getNextNonceApi() {
+    if (null == nextNonceApi) {
+      nextNonceApi = returnNextNonceApi;
+    }
+    return nextNonceApi;
+  }
+
+  public IdexAccountService getAccountService() {
+    if (null == idexAccountService) idexAccountService = new IdexAccountService(this);
+    return idexAccountService;
+  }
+
+  public IdexMarketDataService getMarketDataService() {
+    if (null == idexMarketDataService) idexMarketDataService = new IdexMarketDataService(this);
+    return idexMarketDataService;
+  }
+
+  public IdexTradeService getTradeService() {
+    if (null == idexTradeService) idexTradeService = new IdexTradeService(this);
+    return idexTradeService;
+  }
+
+  @Override
+  protected void initServices() {}
+
+  @Override
+  public SynchronizedValueFactory<Long> getNonceFactory() {
+    return (SynchronizedValueFactory)
+        () -> {
+          Long ret = null;
+          try {
+            ReturnNextNonceResponse var10000 =
+                getNextNonceApi()
+                    .nextNonce(new NextNonceReq().address(getExchangeSpecification().getApiKey()));
+            ret = var10000.getNonce().longValue();
+          } catch (Exception e) {
+            e.printStackTrace();
+          }
+          return ret;
+        };
+  }
+
+  @Override
+  public ExchangeSpecification getDefaultExchangeSpecification() {
+    return new IdexExchangeSpecification();
+  }
+
+  public enum Companion {
+    ;
+    public static final String IDEX_API =
+        System.getProperty("xchange.idex.api", "https://api.idex.market/");
+
+    public static BigDecimal safeParse(String s) {
+      BigDecimal ret = null;
+      try {
+        ret = new BigDecimal(s);
+      } catch (Exception e) {
+      }
+      return ret;
+    }
+
+    public static String getMarket(CurrencyPair currencyPair) {
+      return currencyPair.base.getSymbol() + "_" + currencyPair.counter.getSymbol();
+    }
+
+    public static CurrencyPair getCurrencyPair(String market) {
+      CurrencyPair currencyPair;
+      Iterator<String> syms = asList(market.split("_")).iterator();
+      currencyPair = new CurrencyPair(syms.next(), syms.next());
+      return currencyPair;
+    }
+
+    public static void main(String... args) throws IOException {
```

Comment:
This doesn't belong here, perhaps move it to examples.

Reply:
will revisit this with a testnet example after some discussion.